canal/sync.go

```diff
@@ -101,19 +102,24 @@ func (c *Canal) runSyncBinlog() error {
 			}
 		case *replication.QueryEvent:
 			// handle alert table query
-			if mb := expAlterTable.FindSubmatch(e.Query); mb != nil {
-				if len(mb[1]) == 0 {
-					mb[1] = e.Schema
+			var mb = [][]byte{}
+			// run alter table directly
+			if mb = expAlterTable.FindSubmatch(e.Query); mb == nil {
```

Comment:
@siddontang Sorry, I don't get it clearly. 🙁 
If `continue` in upper condition block(as you said `renameTable()`), it will miss check of alter table event.